### PR TITLE
[Culling] apply frustum culling in Python interfaces

### PR DESCRIPTION
**Pull request by bigbike**
## Motivation and Context

Introduce newly developed frustum culling in python side, so people can use it.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by 1heart** [comment]
# [Codecov](https://codecov.io/gh/facebookresearch/habitat-sim/pull/473?src=pr&el=h1) Report
> Merging [#473](https://codecov.io/gh/facebookresearch/habitat-sim/pull/473?src=pr&el=desc) into [master](https://codecov.io/gh/facebookresearch/habitat-sim/commit/d68db946a16ecfb21bdfffbc2261fe19b1b77d37&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/facebookresearch/habitat-sim/pull/473?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #473   +/-   ##
=======================================
  Coverage   58.99%   58.99%           
=======================================
  Files         165      165           
  Lines        7280     7280           
  Branches       84       84           
=======================================
  Hits         4295     4295           
  Misses       2985     2985           
```

| Flag | Coverage Δ | |
|---|---|---|
| #CPP | `54.26% <0.00%> (ø)` | :arrow_up: |
| #JavaScript | `10.00% <0.00%> (ø)` | :arrow_up: |
| #Python | `79.64% <0.00%> (ø)` | :arrow_up: |


------

[Continue to review full report at Codecov](https://codecov.io/gh/facebookresearch/habitat-sim/pull/473?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/facebookresearch/habitat-sim/pull/473?src=pr&el=footer). Last update [d68db94...d68db94](https://codecov.io/gh/facebookresearch/habitat-sim/pull/473?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/esp/sim/Simulator.h [hunk 1]
```diff
@@ -301,6 +302,18 @@ class Simulator {
...
+   * @brief Enable or disable frustum culling (enabled by default)
+   * @param val, true = enable, false = disable
+   */
+  void setFrustumCulling(bool val) { frustumCulling_ = val; }
+
+  /**
+   * @brief Get status, whether frustum culling is enabled or not
+   * @return true if enabled, otherwise false
+   */
+  bool getFrustumCulling() { return frustumCulling_; }
```
[the remark this answers is not in the document]
Comment (bigbike) [reply src/esp/sim/Simulator.h]: I always use "set/get + noun" as the name of Setter/Getter. 
Yes, your name is better here. Will change.

File: src/esp/gfx/Renderer.cpp [hunk 2]
```diff
@@ -32,22 +32,26 @@ struct Renderer::Impl {
   }
   ~Impl() { LOG(INFO) << "Deconstructing Renderer"; }
 
-  void draw(RenderCamera& camera, scene::SceneGraph& sceneGraph) {
+  void draw(RenderCamera& camera,
+            scene::SceneGraph& sceneGraph,
+            bool frustumCulling) {
```
[the remark this answers is not in the document]
Comment (bigbike) [reply src/esp/gfx/Renderer.cpp]: If I do not put this bool "switch" in the argument list, that means I will have to maintain a "state" called `frustumCulling` in the `Renderer` class, and the same logic also applies to `RenderCamera` class.

Like I said in the previous PR, the calling sequence is `Simulator --> Renderer --> RenderCamera`. Simulator class will own and maintain a copy of this state (`frustumCulling_`) anyway. So I would like to keep it only at the highest level (simulator), avoiding duplicated effort at the lower level, which is too complex and unnecessary.

**Comment by bigbike** [comment]
Run the newly created ab_test.py on randomly picked models on dev-server. Here is the result.
The command:
```
python examples/ab_tests.py --scene ~/models/basis_mp3d/1pXnuDYAj8r/1pXnuDYAj8r.glb --max_frames 4000 --feature frustum_culling -b --control_value False --test_value True
```
Description: 
- The number before `/`: FPS (control group). Frustum culling was disabled;
- The number after `/`: FPS (test group). Frustum culling was enabled;
- The percentage in the parentheses: The performance change ratio (e.g., 40.5% means FPS is improved by 40.5%)

The output:
```
 ====== FPS (512 x 512, depth_only_ABTestGroup.TEST): 4081.3 ======
 ================ Performance (FPS) NPROC=1 ===================================
Resolution  rgb                       rgbd                      depth_only
128 x 128 1165.2/1637.5 (  40.5%) 572.2 /707.8  (  23.7%) 996.0 /1293.8 (  29.9%)
256 x 256 1034.4/1417.2 (  37.0%) 528.9 /658.9  (  24.6%) 904.6 /1249.3 (  38.1%)
512 x 512 818.4 /1041.6 (  27.3%) 399.0 /472.1  (  18.3%) 741.8 /929.5  (  25.3%)
 ==============================================================================
 ================ Performance (FPS) NPROC=3 ===================================
Resolution  rgb                       rgbd                      depth_only
128 x 128 3088.2/4277.4 (  38.5%) 1389.2/1844.3 (  32.8%) 2741.1/3681.4 (  34.3%)
256 x 256 2975.2/3882.7 (  30.5%) 1327.1/1704.2 (  28.4%) 2596.1/3279.3 (  26.3%)
512 x 512 2296.3/2935.6 (  27.8%) 1133.9/1327.4 (  17.1%) 2094.5/2568.1 (  22.6%)
 ==============================================================================
 ================ Performance (FPS) NPROC=5 ===================================
Resolution  rgb                       rgbd                      depth_only
128 x 128 4964.6/6894.1 (  38.9%) 1789.9/2518.8 (  40.7%) 4251.7/5637.6 (  32.6%)
256 x 256 4551.3/6205.3 (  36.3%) 1693.8/2360.3 (  39.3%) 4107.9/5296.6 (  28.9%)
512 x 512 3640.4/4561.8 (  25.3%) 1604.1/1990.8 (  24.1%) 3275.1/4081.3 (  24.6%)
 ==============================================================================
```

**Comment by bigbike** [comment]
Note: I did not touch the `seed` which was using the default value 1.
I assume the agent's `start_state` and the action sequences would be the same in each run.

**Comment by bigbike** [comment]
Dear reviewers,

Let me know if you have further concerns about this diff. Thanks.

**Comment by bigbike** [comment]
Dear reviewers,

I am planning to ship it this week since team needs this functionality.

If it is OK to go, can someone approve this diff? Thanks!

**Pull request merged**